client/components/shipping/packages/input-filters.js

```diff
@@ -0,0 +1,26 @@
+const leftPoint = /^\.\d+$/;
+const rightPoint = /^\d+\.$/;
+const number = ( value ) => {
+	if ( leftPoint.test( value ) ) {
+		value = '0' + value;
+	} else if ( rightPoint.test( value ) ) {
+		value = value.slice( 0, value.length - 1 );
+	}
+
+	return value;
+};
+
+const dimensionRegex = /^(\S+)\s*x\s*(\S+)\s*x\s*(\S+)$/;
+const dimensions = ( value ) => {
+	const result = dimensionRegex.exec( value );
+	if ( result ) {
+		return result.splice( 1, 4 ).map( number ).join( ' x ' );

```

Comment:
I'm really trying to understand this. So, the regex exec will return something like 

`["5 x 6 x 66", "5", "6", "66", index: 0, input: "5 x 6 x 66"]`

and then splice( 1,  4 ) yields

`["5", "6", "66"]`

but that doesn't make sense to me - I thought splice( 1,  4 ) would remove 4 elements from the array starting with the 1st element?

I mean this works, but I don't understand why. Can you explain? :)